### PR TITLE
Add check box "Same in all tools" in the Dynamics options window (fix #4003)

### LINKED ISSUE #4003: Add check box "Same in all Tools" in the Dynamics options window
It would be great to add option to memorize the dynamics options "per tool" or "same for all tools".
Consider saving these preferences between sessions.

Related: https://github.com/aseprite/aseprite/pull/3986

### PR DESCRIPTION
**Pull request by Gasparoken**
Now each tool has independent 'dynamic options'. A global configuration for all tools is also possible via the 'Same in all tools' option.

fix #4003

### CODE REVIEW

**Review by martincapello — APPROVED** [review]

File: src/app/ui/dynamics_popup.cpp [hunk 1]
```diff
@@ -293,9 +295,9 @@ int DynamicsPopup::ditheringIndex() const {
   return 0;
 }
 
-void DynamicsPopup::loadDynamicPref(ToolPreferences* toolPref) {
-  if (toolPref) {
-    auto& dynaPref = toolPref->dynamics;
+void DynamicsPopup::loadDynamicsPref(Preferences* pref) {
+  if (pref) {
+    auto& dynaPref = pref->tool(App::instance()->activeTool()).dynamics;
```
Comment (dacap) [inline src/app/ui/dynamics_popup.cpp:300]: No need to pass the singleton Preferences instance as argument:
```suggestion
void DynamicsPopup::loadDynamicsPref()
{
  auto& dynaPref = Preferences::instance().tool(App::instance()->activeTool()).dynamics;
```

File: src/app/ui/dynamics_popup.h [hunk 2]
```diff
@@ -37,8 +37,11 @@ namespace app {
 
     tools::DynamicsOptions getDynamics() const;
     void setOptionsGridVisibility(bool state);
-    void loadDynamicPref(ToolPreferences* toolPref);
+    void loadDynamicsPref(Preferences* pref);
```
Comment (dacap) [inline src/app/ui/dynamics_popup.h:40]:
```suggestion
    void loadDynamicsPref();
```
And move `#include "app/pref/preferences.h"` to the `dynamics_popup.cpp`

File: src/app/ui/dynamics_popup.h [hunk 3]
```diff
@@ -37,8 +37,11 @@ namespace app {
...
     int ditheringIndex() const;
+    void refreshVisibility();
+    bool sharedSettings() const;
+    void sharedSettings(bool state);
```
Comment (dacap) [inline src/app/ui/dynamics_popup.h:44]: This is not used
```suggestion
```

**Review by dacap — CHANGES_REQUESTED** [review]

**Comment by Gasparoken** [comment]
Ready for review.

File: src/app/ui/context_bar.cpp [hunk 4]
```diff
@@ -1192,6 +1162,27 @@ class ContextBar::DynamicsField : public ButtonSet
       return;
     }
 
+    m_popup.reset(new DynamicsPopup(this));
```
[the remark this answers is not in the document]
Comment (Gasparoken) [reply src/app/ui/context_bar.cpp]: You're right. No needed. I added an `if (!m_popup.get())`

**Comment by Gasparoken** [comment]
> The commit of this PR now looks a little strange ([815ffda](https://github.com/aseprite/aseprite/commit/815ffda1b0c0e2344a490cca86333fae105f74fb)), it looks like @martincapello is the author and you the committer, but I thought you was the author (?)

Incredible! I don't know how that is possible. I'll push again.

**Comment by Gasparoken** [comment]
Done

**Comment by dacap** [comment]
I'll merge it as it is, was able to test it a little, I don't like that checking the "Same in all tools" will set the properties of all tools. I'll make a TODO comment about this.

**Pull request merged**